### DISCUSSION TITLE
glowTTS | gradTTS | VITS (Monotonic Alignment Search MAS)

### DISCUSSION THREAD

**Discussion by p0p4k** [coqui-ai/TTS#2025] [General]
We discussed how to calculate `logp` in [previous discussion](https://github.com/coqui-ai/TTS/discussions/1984).
This article is explanation for the following MAS code: (finding `alignment/path` that maximizes `logp`)
https://github.com/coqui-ai/TTS/blob/d6ad9a05b40e9f3aa876ecfa1c0d23339f373df4/TTS/tts/utils/helpers.py#L199-L238
In this example I assume integer values, but it will be `logp` values in float (since `p` is takes values in [0,1]).

![image](https://user-images.githubusercontent.com/8834712/192745131-b5373c41-52d3-4bbb-864f-9c78bb32af2b.png)
![image](https://user-images.githubusercontent.com/8834712/192745186-3d3c0f0d-e25b-4bdc-b3d0-ec9cf2914726.png)
![image](https://user-images.githubusercontent.com/8834712/192745038-d393b34b-07ef-4754-98f9-b7a4af49c36b.png)

(Previous paper discussions : https://github.com/coqui-ai/TTS/discussions/1494, https://github.com/coqui-ai/TTS/discussions/1501, https://github.com/coqui-ai/TTS/discussions/1506, https://github.com/coqui-ai/TTS/discussions/1799, https://github.com/coqui-ai/TTS/discussions/1984)

**Comment by p0p4k** [comment]
PS. Ignore the 48% in the last image.

**Comment by erogol** [comment]
Awesome content!! Thanks!!
